lib/streamlit/type_util.py

```diff
@@ -455,9 +456,17 @@ def convert_anything_to_df(
     if is_type(df, "numpy.ndarray") and len(df.shape) == 0:
         return pd.DataFrame([])
 
-    if is_type(df, _SNOWPARK_DF_TYPE_STR) or is_type(df, _SNOWPARK_TABLE_TYPE_STR):
-        df = pd.DataFrame(df.take(max_unevaluated_rows))
+    if (
+        is_type(df, _SNOWPARK_DF_TYPE_STR)
+        or is_type(df, _SNOWPARK_TABLE_TYPE_STR)
+        or is_type(df, _PYSPARK_DF_TYPE_STR)
+    ):
+        if is_type(df, _PYSPARK_DF_TYPE_STR):
+            df = df.limit(max_unevaluated_rows).toPandas()
+        else:
+            df = pd.DataFrame(df.take(max_unevaluated_rows))
         if df.shape[0] == max_unevaluated_rows:
+            df._max_unevaluated_rows_caption_was_shown = True
             st.caption(
                 f"⚠️ Showing only {string_util.simplify_number(max_unevaluated_rows)} rows. "
                 "Call `collect()` on the dataframe to show more."
```

Comment:
Does this also apply to pyspark? 

```
"Call `collect()` on the dataframe to show more."
```

If not, it might be better to actually split the pyspark & snowflake sections even if there is some redundancy.

Reply:
`collect()` is valid both for `PySpark` and `Snowpark`, these libraries have `callable` `collect()` method on their DataFrame types 👍